modules/core/src/parallel.cpp

```diff
@@ -101,7 +101,11 @@
     #endif
     #include "tbb/tbb.h"
     #include "tbb/task.h"
+    #if defined(OPENCV_TBB_VERSION_MAJOR) && OPENCV_TBB_VERSION_MAJOR >= 2021
+    #include "tbb/version.h"
```

Comment:
This can be detected through `try_compile()` or checking  `if(EXISTS ...)` / `TBB_VER_FILE`.

Reply:
Thank you. Perhaps `EXIST` check would be the best. I'll update the PR.

Reply:
Looks like we don't these conditional includes at all.
`tbb/version.h` or `tbb/tbb_stddef.h` is included by `tbb/tbb.h` or by `tbb/task.h`:
- latest public release (2021.1): https://github.com/oneapi-src/oneTBB/blame/onetbb_2021/include/oneapi/tbb.h#L71 (`version.h`)
- 2017+: https://github.com/oneapi-src/oneTBB/blame/2017_U1/include/tbb/task.h#L24 (`tbb_stddef.h`)
- 2021-beta05 still uses `tbb/stddef.h`: https://github.com/oneapi-src/oneTBB/commits/88b4c2782a530db1e66947edd2b24699b67850ab/include/tbb/tbb_stddef.h
- 2021-beta08+ uses `version.h`: https://github.com/oneapi-src/oneTBB/blame/v2021.1-beta08/include/tbb/tbb.h#L71

---

P.S. TBB team is not able to work properly with Git history (too many branches is created from clean repo state) =(

Reply:
Removal of these includes works. Tested even on Ubuntu 14.04 with "TBB (ver 4.2 interface 7000)"